Documentation/pre-build-detection

```diff
@@ -0,0 +1,142 @@
+# Pre-build detection guide
```

Comment:
Nit: I think we use pre-buil**t** in most documentation instead of pre-buil**d**. Unless I am missing something here. We should be consistent either way I think.

Reply:
Of course, will do